sdk/iot/core/tests/cmocka/test_az_iot_core.c

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "test_az_iot_core.h"
+#include <az_iot_core.h>
+#include <az_span.h>
+
+#include <az_precondition_internal.h>
+#include <az_precondition.h>
+
+#include <setjmp.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include <cmocka.h>
+#include <az_test_precondition.h>
+
+static void az_span_token_success(void** state)
+{
+  (void)state;
+  az_span span = AZ_SPAN_FROM_STR("abcdefgabcdefgabcdefg");
+  az_span delim = AZ_SPAN_FROM_STR("abc");
+  az_span token;
+  az_span out_span;
+
+  // token: ""
+  token = az_span_token(span, delim, &out_span);
+  assert_non_null(az_span_ptr(token));
+  assert_true(az_span_length(token) == 0);
+  assert_true(az_span_ptr(out_span) == (az_span_ptr(span) + az_span_length(delim)));
+  assert_true(az_span_length(out_span) == (az_span_length(span) - az_span_length(delim)));
+  assert_true(az_span_capacity(out_span) == (az_span_capacity(span) - az_span_capacity(delim)));
+
+  // token: "defg" (span+3)
+  span = out_span;
```

Comment:
The comment seems out of place.
Isn't `span = out_span = defgabcdefgabcdefg` here (i.e. remainder)?

Reply:
Not really, the comment defines that this block is expecting the token "defg". It is not about the span or out_span. We can talk offline to clarify.